### ISSUE TITLE
[7.4.0RC1] test failure and segfault on 32-bit

### ISSUE THREAD

**Issue by remicollet** [krakjoe/componere#23]
Test failure:
```
TEST 87/108 [tests/086.phpt]
========DIFF========
005+ Fatal error: Uncaught Error: Object of class DateTime could not be converted to string in /builddir/build/BUILD/php-componere-3.1.0/ZTS/tests/086.php:37
005- %s fatal error: Object of class DateTime could not be converted to string in %s on line 37
006+ Stack trace:
007+ #0 {main}
008+   thrown in /builddir/build/BUILD/php-componere-3.1.0/ZTS/tests/086.php on line 37
========DONE========
FAIL Patch::derive [tests/086.phpt] 
```

Segfault on 32-bit only
```
TEST 33/108 [tests/032.phpt]
========DIFF========
005+ 
006+ Termsig=11
========DONE========
FAIL Patch with interfaces [tests/032.phpt] 
TEST 34/108 [tests/033.phpt]
========DIFF========
003+ 
004+ Termsig=11
========DONE========
FAIL Patch with interfaces after construction [tests/033.phpt] 

```

**Comment by remicollet** [comment]
test/**032**
```
(gdb) bt
#0  destroy_zend_class (zv=0xffffb60c) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_opcode.c:313
#1  0xf72f3324 in ?? () from /builddir/build/BUILDROOT/php-componere-3.1.0-3.fc29.remi.7.4.i386/usr/lib/php/modules/componere.so
#2  0x56958be6 in zend_objects_store_del (object=0xf707102c) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_objects_API.c:193
#3  0x569207b5 in i_zval_ptr_dtor (zval_ptr=0xffffb6bc) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_variables.h:44
#4  zval_ptr_dtor (zval_ptr=0xffffb6bc) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_variables.c:84
#5  0x569339a8 in _zend_hash_del_el_ex (prev=<optimized out>, p=<optimized out>, idx=8, ht=<optimized out>) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_hash.c:1305
#6  _zend_hash_del_el (p=0xf70562c0, idx=192, ht=0x56f6d668 <executor_globals+168>) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_hash.c:1328
#7  zend_hash_reverse_apply (ht=ht@entry=0x56f6d668 <executor_globals+168>, apply_func=0x569124d0 <zval_call_destructor>, apply_func@entry=0xace674d0)
    at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_hash.c:1899
#8  0x56912895 in shutdown_destructors () at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_execute_API.c:244
#9  0x56922581 in zend_call_destructors () at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend.c:1169
#10 0x568be471 in php_request_shutdown (dummy=0x0) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/main/main.c:1872
#11 0x569af1a3 in do_cli (argc=62, argv=0x56f75680) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/sapi/cli/php_cli.c:1129
#12 0x5678bb09 in main (argc=<optimized out>, argv=<optimized out>) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/sapi/cli/php_cli.c:1352

```
test/**033**
```
(gdb) bt
#0  destroy_zend_class (zv=0xffffb60c) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_opcode.c:313
#1  0xf72f3324 in ?? () from /builddir/build/BUILDROOT/php-componere-3.1.0-3.fc29.remi.7.4.i386/usr/lib/php/modules/componere.so
#2  0x56958be6 in zend_objects_store_del (object=0xf707502c) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_objects_API.c:193
#3  0x569207b5 in i_zval_ptr_dtor (zval_ptr=0xffffb6bc) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_variables.h:44
#4  zval_ptr_dtor (zval_ptr=0xffffb6bc) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_variables.c:84
#5  0x569339a8 in _zend_hash_del_el_ex (prev=<optimized out>, p=<optimized out>, idx=8, ht=<optimized out>) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_hash.c:1305
#6  _zend_hash_del_el (p=0xf70562c0, idx=192, ht=0x56f6d668 <executor_globals+168>) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_hash.c:1328
#7  zend_hash_reverse_apply (ht=ht@entry=0x56f6d668 <executor_globals+168>, apply_func=0x569124d0 <zval_call_destructor>, apply_func@entry=0xace674d0)
    at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_hash.c:1899
#8  0x56912895 in shutdown_destructors () at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend_execute_API.c:244
#9  0x56922581 in zend_call_destructors () at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/Zend/zend.c:1169
#10 0x568be471 in php_request_shutdown (dummy=0x0) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/main/main.c:1872
#11 0x569af1a3 in do_cli (argc=62, argv=0x56f75680) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/sapi/cli/php_cli.c:1129
#12 0x5678bb09 in main (argc=<optimized out>, argv=<optimized out>) at /usr/src/debug/php-7.4.0~RC1-1.fc29.remi.i386/sapi/cli/php_cli.c:1352

```

**Comment by remicollet** [comment]
BTW, fixed in develop branch using 95120639f0bfb6d4a6fe076ddf1b3016eda5e664 and 0f16925652b49bdefaa0ae2264446a845a5bb109
